honeybee_vtk/writer.py

```diff
@@ -0,0 +1,141 @@
+
+            # If base_geometry is found
+            if grids[0]:
+                base_geo_points = get_grid_base(grids[0])[0]
+                write_polydata(base_geo_points, 'grid base')
+                file_names.append('grid base')
+
+            # If base_geometry is not found but mesh faces are found
+            if grids[1]:
+                mesh_points = get_grid_mesh(grids[1])[0]
+                write_polydata(mesh_points, 'grid mesh')
+                file_names.append('grid mesh')
+
+            # If only grid points are found
+            if grids[2]:
+                start_points, end_points, vectors = get_grid_points(grids[2])
+                write_color_grouped_points(start_points, vectors)
+                file_names.append('grid points')
+        else:
+            pass
+        
+        # Write vectors if they are requested
+        if get_vectors:
+            
+            # If Aperture objects are there in HBJSON
+            if 'Aperture' in hb_types:
+                # Create face normals
+                start_points, end_points, normals = get_face_center(
+                    grouped_points['Aperture'])
+                write_arrows(start_points, end_points, normals, 'Aperture')
+                file_names.append('Aperture vectors')
+            
+            # If grids are found in HBJSON
+            if grids:
+                
+                # If base_geometry is found in any of the grids
+                if grids[0]:
+                    base_geo_points = get_grid_base(grids[0])[0]
+                    start_points, end_points, normals = get_face_center(base_geo_points)
+                    write_arrows(start_points, end_points, normals, 'grid base')
+                    file_names.append('grid base vectors')
+
+                # If mesh is found in any of the grids
+                if grids[1]:
+                    mesh_points = get_grid_mesh(grids[1])[0]
+                    start_points, end_points, normals = get_face_center(mesh_points)
+                    write_arrows(start_points, end_points, normals, 'grid mesh')
+                    file_names.append('grid mesh vectors')
+        else:
+            pass
+        
+        # Use the name of HBJSON if file name is not provided
+        if not file_name:
+            name = '.'.join(os.path.basename(file_path).split('.')[:-1])
+            file_name = clean_string(name)
+
+        # Create a .zip file to capture all the generated .vtk files
+        zipobj = ZipFile(file_name + '.zip', 'w')
+        
+        # Capture vtk files in a zip file.
+        for file_name in file_names:
+            zipobj.write(file_name + '.vtk')
+        zipobj.close()
+
+        # Delete vtk files if there is permission
+        for file_name in file_names:
+            try:
+                os.remove(file_name + '.vtk')
+            except OSError:
+                warnings.warn(
+                    'Honeybee does not have permission to delete files. You can delete' 
```

Comment:
I would print the name of the exact vtk file here. It's hard to know which file should be removed.